remoteworkitem/trackerquery_repository.go

```diff
@@ -32,208 +30,128 @@ func NewTrackerQueryRepository(db *gorm.DB) *GormTrackerQueryRepository {
 		}, "tracker query not found")
 
-		return nil, NotFoundError{entity: "TrackerQuery", ID: tq.ID}
-	}
-	if tx.Error != nil {
-		return nil, InternalError{simpleError{fmt.Sprintf("could not load tracker query: %s", tx.Error.Error())}}
-	}
-	trackerRepo := NewTrackerRepository(r.db)
-	err = trackerRepo.CheckExists(ctx, tq.TrackerID)
-	if err != nil {
-		switch err.(type) {
-		case errors.NotFoundError:
-			log.Error(ctx, map[string]interface{}{
-				"err":        err,
-				"tracker_id": id,
-			}, "tracker not found by its ID")
-			return nil, NotFoundError{entity: "tracker", ID: tq.TrackerID.String()}
-		default:
-			return nil, InternalError{simpleError{fmt.Sprintf("could not load tracker: %s", tx.Error.Error())}}
-		}
+		return nil, errors.NewNotFoundError("TrackerQuery", tq.ID.String())
 	}
 
-	newTq := TrackerQuery{
-		ID:        id,
-		Schedule:  tq.Schedule,
-		Query:     tq.Query,
-		TrackerID: tq.TrackerID,
-		SpaceID:   *tq.Relationships.Space.Data.ID,
+	tx = r.db.Where("tracker_id = ?", tq.TrackerID).Find(&res)
+	if tx.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"err":        tx.Error,
+			"tracker_id": tq.TrackerID,
+		}, "tracker ID not found")
+		return nil, errors.NewNotFoundError("Tracker", tq.TrackerID.String())
 	}
 
-	if err := tx.Save(&newTq).Error; err != nil {
+	if err := tx.Save(&res).Error; err != nil {
 		log.Error(ctx, map[string]interface{}{
-			"err":           err,
-			"tracker_query": tq.Query,
-			"tracker_id":    tq.TrackerID,
+			"trackerquery_id": tq.ID,
+			"err":             err,
 		}, "unable to save the tracker query")
-		return nil, InternalError{simpleError{err.Error()}}
+		return nil, errors.NewInternalError(ctx, err)
 	}
 
-	log.Info(ctx, map[string]interface{}{
-		"tracker_query": newTq,
-	}, "Updated tracker query")
-
-	spaceSelfURL := rest.AbsoluteURL(goa.ContextRequest(ctx).Request, app.SpaceHref(tq.Relationships.Space.Data.ID.String()))
-	t2 := app.TrackerQuery{
-		ID:        tq.ID,
-		Schedule:  tq.Schedule,
-		Query:     tq.Query,
-		TrackerID: tq.TrackerID,
-		Relationships: &app.TrackerQueryRelationships{
-			Space: app.NewSpaceRelation(*tq.Relationships.Space.Data.ID, spaceSelfURL),
-		},
-	}
-
-	return &t2, nil
+	return &res, nil
 }
 
 // Delete deletes the tracker query with the given id
 // returns NotFoundError or InternalError
-func (r *GormTrackerQueryRepository) Delete(ctx context.Context, ID string) error {
-	var tq = TrackerQuery{}
-	id, err := strconv.ParseUint(ID, 10, 64)
-	if err != nil || id == 0 {
-		// treat as not found: clients don't know it must be a number
-		return NotFoundError{entity: "trackerquery", ID: ID}
-	}
-	tq.ID = id
-	tx := r.db
-	tx = tx.Delete(tq)
-	if err = tx.Error; err != nil {
-		return InternalError{simpleError{err.Error()}}
+func (r *GormTrackerQueryRepository) Delete(ctx context.Context, ID uuid.UUID) error {
+	defer goa.MeasureSince([]string{"goa", "db", "trackerquery", "delete"}, time.Now())
+	if ID == uuid.Nil {
```

Comment:
You don't check for Nil when creating a tracker query (https://github.com/fabric8-services/fabric8-wit/pull/2351/files#diff-cbc645207e5431802cb2869a92ce9197R43). It seems that this is not consistent with this check. Please add a test to see what happens. I'm not entirely sure a new UUID will be generated when it is 0000-0000-0000-..... . But I guess it should.

Reply:
I don't check that trackerquery ID is Nil when creating a tracker query because we don't pass tracker query ID in payload for create action

Reply:
But you could technically pass it during creation time, no? I mean you just pass a TQ object and that has an ID.

Reply:
@kwk check this 
c69e98b